runtime/io.js

```diff
@@ -80,11 +71,27 @@ function caml_sys_open (name, flags, _perms) {
   var root = resolve_fs_device(name);
   var file = root.device.open(root.rest,f);
   var idx = caml_global_data.fd_last_idx?caml_global_data.fd_last_idx:0;
-  return caml_sys_open_internal (idx+1,caml_std_output,file,f);
-}
-caml_sys_open_internal(0,caml_std_output, new MlFakeFile(caml_create_bytes(0)), {rdonly:1}); //stdin
-caml_sys_open_internal(1,js_print_stdout, new MlFakeFile(caml_create_bytes(0)), {buffered:2}); //stdout
-caml_sys_open_internal(2,js_print_stderr, new MlFakeFile(caml_create_bytes(0)), {buffered:2}); //stderr
+  return caml_sys_open_internal (idx+1,file,f);
+}
+(function () {
+  function file(fd, name) {
+    if(fs_node_supported()) {
+      try {
+        var fs = require("fs");
+        var fd2 = fs.openSync(name, "rs");
+        new MlNodeFile(fd2, true);
+      } catch(e) {
+        new MlNodeFile(fd, true)
+      }
+      return new MlNodeFile(fd, true)
```

Comment:
This won't work on Windows. Windows stdin, stdout, and sterr are on different file descriptors

Reply:
Here is what I read from the nodejs documentation, is this incorrect ?
```
process.stdin.fd <number>
This property refers to the value of underlying file descriptor of process.stdin. The value is fixed at 0.

process.stdout.fd <number>
This property refers to the value of underlying file descriptor of process.stdout. The value is fixed at  1.

process.stderr.fd <number>
This property refers to the value of underlying file descriptor of process.stderr. The value is fixed at 2. 
```

Reply:
Also note that the CI run with windows as well. It seems to work on windows with the setup used in the CI.

Reply:
When I was investigating Windows support for Grain, Windows didn't use `0`, `1`, and `2`, so I think the nodejs documentation is wrong.